### PR TITLE
Update sample app

### PR DESCRIPTION
**Pull request by geof90**
This PR updates the sample app to use React Native 0.30 so that we can put new features in it e.g. decorator support. It also removes the old tests that are not in use anymore.

### CODE REVIEW

File: Examples/CodePushDemoApp/iOS/CodePushDemoApp/Info.plist [hunk 1]
```diff
@@ -15,18 +15,13 @@
 	<key>CFBundlePackageType</key>
 	<string>APPL</string>
 	<key>CFBundleShortVersionString</key>
-	<string>1.2.3</string>
```
Comment (silhouettes) [inline Examples/CodePushDemoApp/iOS/CodePushDemoApp/Info.plist]: Is this change intentional?

File: Examples/CodePushDemoApp/iOS/CodePushDemoApp/AppDelegate.m [hunk 2]
```diff
@@ -1,59 +1,25 @@
...
-  /**
-   * OPTION 2
-   * Load from pre-bundled file on disk. To re-generate the static bundle
-   * from the root of your project directory, run
-   *
-   * $ react-native bundle --minify
-   *
-   * see http://facebook.github.io/react-native/docs/runningondevice.html
-   */
-
```
Comment (silhouettes) [inline Examples/CodePushDemoApp/iOS/CodePushDemoApp/AppDelegate.m]: Did you mean to remove these instructions?

File: Examples/CodePushDemoApp/android/app/build.gradle [hunk 3]
```diff
@@ -73,27 +81,27 @@ def enableSeparateBuildPerCPUArchitecture = false
...
     buildToolsVersion "23.0.1"
 
     defaultConfig {
-        applicationId "com.microsoft.codepushdemoapp"
+        applicationId "com.codepushdemoapp"
         minSdkVersion 16
         targetSdkVersion 22
         versionCode 1
-        versionName "1.0.0"
+        versionName "1.0"
```
Comment (silhouettes) [inline Examples/CodePushDemoApp/android/app/build.gradle]: semver, here and in `AndroidManifest.xml'?

**Comment by silhouettes** [comment]
Looks good! Might talk to you about the tests later

File: Examples/CodePushDemoApp/iOS/CodePushDemoApp/AppDelegate.m (hunk 2 above)
Comment (geof90) [inline Examples/CodePushDemoApp/iOS/CodePushDemoApp/AppDelegate.m]: Yes, because with RCTBundleURLProvider they no longer apply

**Comment by geof90** [comment]
@scottbommarito already checked in tests using a different mechanism, so these demo app tests are no longer used.

File: Examples/CodePushDemoApp/iOS/CodePushDemoApp/AppDelegate.m (hunk 2 above)
Comment (silhouettes) [inline Examples/CodePushDemoApp/iOS/CodePushDemoApp/AppDelegate.m]: If you don't mind, could you add a little explanation on what uncommenting the top line does? Does it just disable CodePush and load from the development server?

Comment (geof90) [inline Examples/CodePushDemoApp/iOS/CodePushDemoApp/AppDelegate.m]: Done
